.github/workflows/unit_test.yaml

```diff
@@ -1 +1,30 @@
 name: unit test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  unit_tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+      - name: Setup conda env
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniconda-version: "latest"
+          activate-environment: test
+          python-version: "3.8"
+      - name: Install dependencies
+        shell: bash -l {0}
+        run: |
+          conda activate test
```

Comment:
How about adding `set -eux` for all job steps that contain bash commands? I've seen this a lot in automation (e.g., [pytorch](https://github.com/search?q=org%3Apytorch+%22set+-eux%22&type=Code)) which guarantees that we will immediately exist the shell when an error occurs. Since each job step generates a new process and a new shell, we will need to add it for all of them.